File: students/students/src/MON.SPA/src/models/equalizationModel.js
import { DocumentModel } from './documentModel.js';
export class EqualizationModel {
  constructor(obj = {}) {
    this.id = obj.id,
    this.personId = obj.personId;
    this.reasonId = obj.reasonId;
    this.reasonName = obj.reasonName;
    this.inClass = obj.inClass || "";
    this.basicClassName = obj.basicClassName;
    this.equalizationDetails = obj.equalizationDetails || [];
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
    this.sortOrder = obj.sortOrder;
    this.schoolYear = obj.schoolYear;
    this.schoolYearName = obj.schoolYearName;
    this.horarium = obj.horarium;
    this.sessionId = obj.sessionId || '';
  }
}

File: students/students/src/MON.Models/EqualizationModel.cs
namespace MON.Models
{
    using System.Collections.Generic;

    public class EqualizationModel
    {
        public int Id { get; set; }

        public int PersonId { get; set; }

        public int ReasonId { get; set; }

        //Ако причината е: преместване на ученика от VIII до XII клас вкл. (чл. 32, ал. 1, т. 1 от Наредба №11 - трябва да запишем и класа
        public int? InClass { get; set; }

        public IEnumerable<EqualizationDetailsModel> EqualizationDetails { get; set; }
        public IEnumerable<DocumentViewModel> Documents { get; set; }

        public short? SchoolYear { get; set; }
        public string SchoolYearName { get; set; }

        public string ReasonName { get; set; }
        public string BasicClassName { get; set; }


    }
}
